test/e2e/e2e_test.go

```diff
@@ -222,6 +224,103 @@ func TestE2EFlow(t *testing.T) {
+		podName := "test-useraccount-delete-1"
+		laraSignUp := CreateAndApproveSignup(t, hostAwait, laraUserName, memberAwait.ClusterName)
+		VerifyResourcesProvisionedForSignup(t, hostAwait, laraSignUp, "base", memberAwait)
+
+		// when
+		//Create a pod with finalizer in user's namespace, which will block the deletion of namespace.
+		memberPod := corev1.Pod{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Pod",
+				APIVersion: "v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      podName,
+				Namespace: userNamespace,
+				Finalizers: []string{
+					"kubernetes",
+				},
+			},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{{
+					Name:  "test",
+					Image: "busybox",
+				}},
+			},
+		}
+		err = memberAwait.Client.Create(context.TODO(), &memberPod)
+		require.NoError(t, err)
+
+		// confirm pod created
+		pod := &corev1.Pod{}
+		err = memberAwait.Client.Get(context.TODO(), types.NamespacedName{Namespace: userNamespace, Name: podName}, pod)
+		require.NoError(t, err)
+
+		deletePolicy := metav1.DeletePropagationForeground
+		deleteOpts := &client.DeleteOptions{
+			PropagationPolicy: &deletePolicy,
+		}
+		// now delete userSignup but nothing should be deleted yet
+		err = hostAwait.Client.Delete(context.TODO(), laraSignUp, deleteOpts)
+		require.NoError(t, err)
+
+		//time.Sleep(time.Second * 20)
+		templateRefs := tiers.GetTemplateRefs(hostAwait, "base")
+		requiredRef := ""
+		for _, ref := range templateRefs.Namespaces {
+			if strings.Contains(ref, "dev") {
+				requiredRef = ref
+			}
+		}
+		require.True(t, len(requiredRef) > 0, "did not find the required templateRef")
+		_, err = memberAwait.WithRetryOptions(wait.TimeoutOption(time.Second*10)).WaitForNamespace(laraUserName, requiredRef)
+		require.NoError(t, err)
+
+		nsTmplSet := &toolchainv1alpha1.NSTemplateSet{}
+		err = memberAwait.Client.Get(context.TODO(), types.NamespacedName{Namespace: memberAwait.Namespace, Name: laraUserName}, nsTmplSet)
+		require.NoError(t, err)
+
+		userAcc := &toolchainv1alpha1.UserAccount{}
+		err = memberAwait.Client.Get(context.TODO(), types.NamespacedName{Namespace: memberAwait.Namespace, Name: laraUserName}, userAcc)
+		require.NoError(t, err)
+
+		mur := &toolchainv1alpha1.MasterUserRecord{}
+		err = hostAwait.Client.Get(context.TODO(), types.NamespacedName{Namespace: hostAwait.Namespace, Name: laraUserName}, mur)
+		require.NoError(t, err)
+
+		err = hostAwait.Client.Get(context.TODO(), types.NamespacedName{Namespace: laraSignUp.Namespace, Name: laraSignUp.Name}, laraSignUp)
+		require.NoError(t, err)
+
+		// now remove finalizer from pod and check all are deleted
+		err = memberAwait.Client.Get(context.TODO(), types.NamespacedName{Namespace: userNamespace, Name: podName}, pod)
+		require.NoError(t, err)
+		require.Equal(t, pod.Finalizers[0], "kubernetes")
+		pod.Finalizers = nil
+		err = memberAwait.Client.Update(context.TODO(), pod)
+		require.NoError(t, err)
+		require.Equal(t, len(pod.Finalizers), 0)
```

Comment:
It's safer to update resources in a pool function to avoid possible conflicts if the resource (the pod in this case) has been changed between `Get()` and `Update()` but some controller in the cluster. Otherwise we could create another flaky test.
We what we use for modifying MUR for example: https://github.com/codeready-toolchain/toolchain-e2e/blob/8592e7778727ac79295a76c8ef126f278f333e92/testsupport/wait/host.go#L117
For example we can introduce something like:
```
func (a *MemberAwaitility) UpdatePod(namespace, podName string, modifyPod func(pod *corev1.Pod)) (*corev1.Pod, error) {
...
}
```
And use it:
```
err = memberAwait.UpdatePod(namespace, podName, func(pod *corev1.Pod) {
    pod.Finalizers[0] = nil
})
```